### PR TITLE
feat: Support authentication from Auth Action

### PR DESCRIPTION
**Pull request by bharathkkb**
This adds support for discovering `GOOGLE_GHA_CREDS_PATH` set by Auth(https://github.com/google-github-actions/auth/pull/57) and WIF credentials file.

This is backwards compatible allowing other gcloud Actions to support both explicit credentials and authentication via Auth Action.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ import * as downloadUtil from './download-util';
 import * as installUtil from './install-util';
 import { getLatestGcloudSDKVersion } from './version-util';
 import { ExecOptions } from '@actions/exec/lib/interfaces';
+import * as fs from 'fs';
```
Comment (sethvargo) [inline src/index.ts:25]: It looks like we're only using one `fs` method and it's from promises. Can we do `import { promises as fs } from 'fs'` instead?

File: src/index.ts [hunk 2]
```diff
@@ -174,12 +175,60 @@ export function parseServiceAccountKey(
...
+ * Authenticates the gcloud tool using a service account key or WIF credential configuration
+ * discovered via GOOGLE_GHA_CREDS_PATH environment variable. An optional serviceAccountKey
+ * param is supported for legacy Actions.
  *
  * @param serviceAccountKey - The service account key used for authentication.
  * @returns exit code.
  */
 export async function authenticateGcloudSDK(
+  serviceAccountKey?: string,
+  silent = true,
```
Comment (sethvargo) [inline src/index.ts:202]: What does `silent` do?

File: src/test-util.ts [hunk 3]
```diff
@@ -56,6 +59,18 @@ export class TestToolCache {
   }
 }
 
+export const TEST_TMP_FILES_DIR = path.join(os.tmpdir(), 'test-setupcloudsdk');
+
+/**
+ * Helper to write test data to disk.
+ */
+export async function writeTmpFile(data: string): Promise<string> {
```
Comment (sethvargo) [inline src/test-util.ts:67]: We probably want to use `writeSecureFile` instead: https://github.com/google-github-actions/auth/blob/9a051ab76ce8a17efc6be51a4f66a0248970f317/src/utils.ts#L13

Comment (sethvargo) [reply src/test-util.ts:67]: That will also drop the dep on uuid

File: src/index.ts (hunk 1 above)
Comment (bharathkkb) [reply src/index.ts]: done

File: src/index.ts (hunk 2 above)
Comment (bharathkkb) [reply src/index.ts:203]: Added to doc. It's exposing [this option](https://github.com/actions/toolkit/blob/e9b0746ee3e4a146656b69f5dcdd9946a17e8d5c/packages/exec/src/interfaces.ts#L13).

File: src/test-util.ts (hunk 3 above)
Comment (bharathkkb) [reply src/test-util.ts:67]: Good idea, reused that and removed uuid.

**Review by sethvargo — APPROVED** [review]

**Review by averikitsch — APPROVED** [review]

**Pull request merged**